core/driver/sources/src/main/scala/org/virtuslab/ideprobe/dependencies/DependencyResolver.scala

```diff
@@ -3,3 +3,9 @@ package org.virtuslab.ideprobe.dependencies
 trait DependencyResolver[Key] {
   def resolve(key: Key): Dependency
 }
+
```

Comment:
Why is this needed? We can simply pass function where such trait is expected anyway.

Reply:
Isn't needed per se... but makes it much more clear than we construct an instance of `DependencyResolver` 😅 

I was confused when saw

```scala
def resolver(artifact: String): DependencyResolver[IntelliJVersion] = { version =>
....
  Dependency(artifactWithReplacementsApplied)
}
```

Didn't notice the `version =>` and though that `def resolver` returns a `Dependency`, not a `DependencyResolver`